### PR TITLE
Port IntSet to new iterator protocol

### PR DESCRIPTION
**Pull request by quinnj**
@Evizero

### CODE REVIEW

**Comment by quinnj** [comment]
@StephenVavasis, have you worked at all on porting all the Sorted structures to use the new iteration protocol? That's the big remaining chunk left to get DataStructures ready for 1.0 (release imminent). Please let us know if you can do that today or we will probably do it here at JuliaCon.

File: src/int_set.jl [hunk 1]
```diff
@@ -26,9 +26,9 @@ sizehint!(s::IntSet, n::Integer) = (_resize0!(s.bits, n+1); s)
 
 # only required on 0.3:
 function first(itr::IntSet)
-    state = start(itr)
-    done(itr, state) && throw(ArgumentError("collection must be non-empty"))
-    next(itr, state)[1]
+    state = iterate(itr)
```
Comment (andreasnoack) [inline src/int_set.jl]: This method is causing an error and should just be removed since there is a fallback that will work here.

Comment (quinnj) [reply src/int_set.jl]: Yes, I have a fix for this locally

**Comment by JeffBezanson** [comment]
Needs this:

```diff
--- a/src/int_set.jl
+++ b/src/int_set.jl
@@ -28,7 +28,7 @@ sizehint!(s::IntSet, n::Integer) = (_resize0!(s.bits, n+1); s)
 function first(itr::IntSet)
     state = iterate(itr)
     state === nothing && throw(ArgumentError("collection must be non-empty"))
-    iterate(itr, state[2])[1]
+    state[1]
 end
```

**Comment by JeffBezanson** [comment]
Oops, Andreas beat me by 8 minutes :)

**Comment by quinnj** [comment]
Pushed a few fixes; still working on fixing the EnumerateAll iteration

**Comment by StephenVavasis** [comment]
@quinnj in response to your question posted above, yes, #384 updates the iteration protocol for the sorted containers.  It stalled because it did not pass CI.  The coverage went down too much.  If you want me to work on it again, let me know.  Except for more recent changes to master that may be incompatible, it should be ready to merge.

**Comment by quinnj** [comment]
I just pushed my version that passes tests locally; does that look ok? There's some kind of issue w/ ordered_dict tests I'm working through now.

**Comment by JeffBezanson** [comment]
This should fix OrderedDicts:

```diff
--- a/src/ordered_dict.jl
+++ b/src/ordered_dict.jl
@@ -141,7 +141,7 @@ function rehash!(h::OrderedDict{K,V}, newsz = length(h.slots
 
     if h.ndel > 0
         ndel0 = h.ndel
-        ptrs = !isbits(K)
+        ptrs = !isbitstype(K)
         to = 1
```

**Comment by quinnj** [comment]
@shashi for president!

**Comment by Evizero** [comment]
nice

**Comment by quinnj** [comment]
Yes, we're done for the moment; we tagged a new release that will work on 1.0.

**Pull request merged**